src/utils/sanitizer2.ts

```
@@ -0,0 +1,593 @@
+
+const REF_LINK_DEF =
+    /^\s*\[([^\]]{1,100})\]:\s*(?:<([^>\n]{1,2000})>|([^\s]+))(?:\s+(?:"([^"]{0,500})"|'([^']{0,500})'|\(([^()]{0,500})\)))?\s*$/gm;
+
+const HTML_COMMENTS = /<!--[\s\S]*?-->/g;
+
+const GLOBAL_SCHEME = /((^|[\s"'(>]))(?:javascript|vbscript|data)\s*:/gi;
+
+// Controls per §3.1
+const RE_BIDI_INVIS =
+    /[\u202A-\u202E\u2066-\u2069\u200E\u200F\u061C\u00AD\u2060\uFEFF\u200B-\u200D]/g;
+const RE_C0C1_EXCEPT_NL = /[\x00-\x08\x0B\x0C\x0E-\x1F\x7F-\x9F]/g;
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)